### PR TITLE
feat: remove corrupted file on error

### PR DESCRIPTION
**Pull request by OlivierDehaene**

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -14,7 +16,16 @@ fn download(url: String, filename: String, max_files: usize, chunk_size: usize)
     tokio::runtime::Builder::new_multi_thread()
         .enable_all()
         .build()?
-        .block_on(async { download_async(url, filename, max_files, chunk_size).await })
+        .block_on(async { download_async(url, filename.clone(), max_files, chunk_size).await })
+        .map_err(|err| {
+            let path = Path::new(&filename);
+            if path.exists() {
+                remove_file(filename)
+                    .map_err(|err| panic!("Error while removing corrupted file: {:?}", err))
```
Comment (OlivierDehaene) [inline src/lib.rs]: I panic here which is not great... Is there a way to chain exceptions in PyO3? I couldn't find anything.

Comment (Narsil) [reply src/lib.rs]: What do you mean chain ? `PyException::new_err(string)` to create a new Python exception.

Comment (OlivierDehaene) [reply src/lib.rs]: I mean that this exception will happen while handling another exception. In python you would see something like: 

```python
Traceback (most recent call last):
  File "<stdin>", line 2, in <module>
ValueError: first error

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "<stdin>", line 4, in <module>
ValueError: second error
```

I don't know if we have the same behaviour here.

Comment (Narsil) [reply src/lib.rs]: Oh, that I don't know.

Comment (Narsil) [reply src/lib.rs]: For warnings I had to manually call the python runtime through rust. Not sure about chained exceptions.

Comment (Narsil) [reply src/lib.rs]: https://github.com/huggingface/safetensors/pull/182/files#diff-57af18dbd0070e35a14e183a52ece76c980f7c45f3dd3835cfacfcc0669a9884R492

File: Cargo.toml [hunk 2]
```diff
@@ -11,5 +11,5 @@ crate-type = ["cdylib"]
 [dependencies]
 futures = "0.3.25"
 pyo3 = { version = "0.17.3", features = ["extension-module"] }
-reqwest = "0.11.13"
+reqwest = { version = "0.11.13", features = ["rustls-tls"] }
```
Comment (Narsil) [inline Cargo.toml]: Ah ? Porque ?

Comment (OlivierDehaene) [reply Cargo.toml]: Just trying to make the linux CI build happy.

File: Cargo.toml [hunk 3]
```diff
@@ -10,6 +10,7 @@ crate-type = ["cdylib"]
 
 [dependencies]
 futures = "0.3.25"
+openssl = { version = "0.10", features = ["vendored"] }
```
Comment (OlivierDehaene) [inline Cargo.toml]: Static link openssl

File: src/lib.rs (hunk 1 above)
Comment (OlivierDehaene) [reply src/lib.rs]: Do you preffer I change to a warning or this is fine?

File: Cargo.toml (hunk 3 above)
Comment (Narsil) [reply Cargo.toml]: I'd rather we let users choose their openssl so they can upgrade on their own, imo this is better for a lib.

Couldn't we just install ssl before in the CI ?

File: src/lib.rs (hunk 1 above)
Comment (Narsil) [reply src/lib.rs]: It's ok for now. This package is not stable in any form anyway.

**Review by Narsil — APPROVED** [review]

**Pull request merged**